### ISSUE TITLE
React Native, code push update remove previous images

### ISSUE THREAD

**Issue by dkahdwk** [microsoft/react-native-code-push#2136]
### Steps to Reproduce
1. Apply code push to the react-native app.
2. Publish Google Play Store after app build.
3. I can see the image after installation in Google Play. 
4. But image not visible after code push on physical device(android).
5. But the virtual simulator shows the image.

* Image form: ex) <Image source={require('@assets/app_logo.png')} />
* Image location: https://github.com/dkahdwk/travel/tree/main/assets
* https://github.com/facebook/react-native/issues/13600#issuecomment-296192734 -> It didn't work for me.

No matter how hard I try, I don't know the solution!!!

### Reproducible Demo

* https://github.com/dkahdwk/travel/

### Environment

* react-native-code-push version: 7.0.2
* react-native version: 0.64.2
* iOS/Android/Windows version: physical device) Android galaxy s8+
* Does this reproduce on a debug build or release build? yes
* Does this reproduce on a simulator, or only on a physical device? physical device

(The more info the faster we will be able to address it!)

**Comment by rukmanary** [comment]
In our case, first time we codepush in the same target version is works fine. But the problem appears in the second codepush. The Image asset in our app cannot loaded. So I ask my co-worker who implement this package too and he found something in the comment but he forgot which git issue. The solution from that comment is to modify some code in react-native-code-push... Here's a patch that he gives to me:

```
diff --git a/node_modules/react-native-code-push/android/codepush.gradle b/node_modules/react-native-code-push/android/codepush.gradle
index 51e2af3..2769432 100644
--- a/node_modules/react-native-code-push/android/codepush.gradle
+++ b/node_modules/react-native-code-push/android/codepush.gradle
@@ -17,7 +17,7 @@ void runBefore(String dependentTaskName, Task task) {
     }
 }
 
-gradle.projectsEvaluated {
+afterEvaluate {
     android.buildTypes.each {
         // to prevent incorrect long value restoration from strings.xml we need to wrap it with double quotes
         // https://github.com/microsoft/cordova-plugin-code-push/issues/264
@@ -62,6 +62,24 @@ gradle.projectsEvaluated {
                 enabled config."bundleIn${targetName}" ||
                 config."bundleIn${variant.buildType.name.capitalize()}" ?:
                 targetName.toLowerCase().contains("release")
+            doLast {
+                    copy {
+                        into("${buildDir}/intermediates")
+                        into ("assets/${targetPath}") {
+                            from(jsBundleDir)
+                        }
+
+                        // Workaround for Android Gradle Plugin 3.2+ new asset directory
+                        into ("merged_assets/${variant.name}/merge${targetName}Assets/out") {
+                            from(jsBundleDir)
+                        }
+
+                        // Workaround for Android Gradle Plugin 3.4+ new asset directory
+                        into ("merged_assets/${variant.name}/out") {
+                            from(jsBundleDir)
+                        }
+                    }
+                }   
             }
         } else {
             def jsBundleDirConfigName = "jsBundleDir${targetName}"
```

I hope this helpful

**Comment by dkahdwk** [comment]
<img width="400" alt="스크린샷 2021-09-01 오후 7 26 48" src="https://user-images.githubusercontent.com/56759331/131655977-f879bf2b-7816-4b58-b948-01c0826bebf2.png">

First of all, thank you for your answer!
I added the code after hearing your answer, but the problem was not solved.
I'd appreciate it if you could check it out!

**Comment by aragorndp1812** [comment]
here is the solution, edit file CodePush.js in node_modules,
func: async function checkForUpdate(deploymentKey = null, handleBinaryVersionMismatchCallback = null) {

  var update = await sdk.queryUpdateWithCurrentPackage(queryPackage);

  if(update != null){
    queryPackage = { appVersion: config.appVersion };
    update = await sdk.queryUpdateWithCurrentPackage(queryPackage);
  }

**Comment by HagarAyad** [comment]
I'am using React native v 0.63.4 and react-native-code-push version: 7.0.2 and have same issue on physical devices,
when downgrade react-native-code-push version to 7.0.1 ,it works fine

**Comment by dkahdwk** [comment]
> 나는 React native v 0.63.4 및 react-native-code-push 버전: 7.0.2를 사용하고 있으며 물리적 장치에서 동일한 문제가 있습니다.
> react-native-code-push 버전을 7.0.1로 다운그레이드하면 잘 작동합니다.

I lowered the version to 7.0.1 but it still didn't work...

**Comment by Andriiklymiuk** [comment]
> I'am using React native v 0.63.4 and react-native-code-push version: 7.0.2 and have same issue on physical devices,
> when downgrade react-native-code-push version to 7.0.1 ,it works fine

@HagarAyad thanks a lot, you are life saver!
Your advice helped on 0.65.1 react native

**Comment by AndrewShapovalov** [comment]
Can confirm that downgrading to the version `7.0.1` resolves the issue.

**Comment by dkahdwk** [comment]
> Can confirm that downgrading to the version `7.0.1` resolves the issue.

Others say it's all solved by versioning down to 7.0.1 but I can't... The 6.2.0 version was also attempted, but it was not resolved. What's my problem?

**Comment by Andriiklymiuk** [comment]
My guess it is either you haven't:
1. deleted previous build and installed new build with 7.0.2
2. run `yarn install` and your yarn.lock isn't updated

Also, look into installation instructions. For me, 0.65.1 rn pushed for complete project reinitialisation, maybe it can also help

**Comment by dkahdwk** [comment]
> My guess it is either you haven't:
> 
> 1. deleted previous build and installed new build with 7.0.1
> 2. run `yarn install` and your yarn.lock isn't updated
> 
> Also, look into installation instructions. For me, 0.65.1 rn pushed for complete project reinitialisation, maybe it can also help

I forgot to rebuild it. Thank you!

**Comment by fbartho** [comment]
I'm exhibiting this issue as well.

**Comment by alexandergoncharov** [comment]
Hi there :)
We merged and published this PR: https://github.com/microsoft/react-native-code-push/pull/2149

Could you please try to reproduce your issue with a new version and let us know if it was successful or not?

**Comment by ianitararache** [comment]
Unfortunately it is not fixed with v7.0.4

**Comment by MufeedaFi** [comment]
Can we get to know from where the images in React native is being loaded from natively in android? ie, in android native side, where should we look up for the images?
Or is there any way we can override the resources used in native android side for a React native project like the below function that overrides the bundle file?

@OverRide
protected String getJSBundleFile() {
